### ISSUE TITLE
useReanimatedKeyboardAnimation is not animating the progress value

### ISSUE THREAD

**Issue by itsramiel** [kirillzyusko/react-native-keyboard-controller#85]
**Describe the bug**
The progress shared value returned from useReanimatedKeyboardAnimation does not progress the value from 0 to 1

**Code snippet**
```js
const Example = () => {
  const {progress, height} = useReanimatedKeyboardAnimation();

  const rStyle = useAnimatedStyle(() => {
    return {
      backgroundColor: 'gray',
      height: Math.abs(height.value),
      width: interpolate(progress.value, [0, 1], [100, 200]),
    };
  }, []);

  return (
    <SafeAreaView
      style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
      <TextInput
        style={{
          borderWidth: 1,
          padding: 10,
          borderColor: 'brown',
          width: '100%',
        }}
      />
      <Button title="dismiss" onPress={() => Keyboard.dismiss()} />
      <Animated.View style={rStyle} />
    </SafeAreaView>
  );
};
```

**Repo for reproducing**
https://github.com/itsramiel/animated-keyboard

**Expected behavior**
progress shared value do not suddenly switch between 0 and 1, but progress the value between the range 0 and 1 as the keyboard opens

**Screenshots**

https://user-images.githubusercontent.com/80689446/194716302-664a0092-50a6-4dee-9210-fe187cba02e9.mp4

**Smartphone (please complete the following information):**
 - Desktop OS: [MacOS 
 - Device: iiPhone8 plus
 - OS: iOS 16
 - RN version: 0.70.x
 - RN architecture: [e.g. old/new or paper/fabric]
 - JS engine: Hermes
 - Library version: 1.3.0

**Comment by kirillzyusko** [comment]
Hi @itsramiel 
Thank you for the issue. This problem does indeed exist and it's pointed out in documentation:

<img width="884" alt="image" src="https://user-images.githubusercontent.com/22820318/194719231-a40b3959-8952-4062-a81a-b61376be7dc2.png">

I'm working on a solution and hopefully in next upcoming release (`1.4.0`) it will be fixed 🤞

**Comment by itsramiel** [comment]
Thank you for the response!! I think reanimated released a very similar thing?

**Comment by kirillzyusko** [comment]
@itsramiel yes, you are right. Reanimated released a similar hook, though our implementations have some differences.

But in my opinion keyboard is a more complex thing, than just one hook, so I'm still going to support this library and add new features into it.

But if you can use hook from `reanimated` and it fits your purposes - go ahead with it 🙂

**Comment by itsramiel** [comment]
appreciate your effort 🚀

**Comment by matthieugicquel** [comment]
> Hi [@itsramiel](https://github.com/itsramiel) Thank you for the issue. This problem does indeed exist and it's pointed out in documentation:
> 
> <img alt="image" width="884" src="https://user-images.githubusercontent.com/22820318/194719231-a40b3959-8952-4062-a81a-b61376be7dc2.png">
> 
> I'm working on a solution and hopefully in next upcoming release (`1.4.0`) it will be fixed 🤞

@kirillzyusko It looks like this mention is not present in the docs anymore, even though it's still true for the values `useReanimatedKeyboardAnimation` isn't it?

I've seen that it can successfully animate things when used in an animated style, but when trying to perform side effects (eg console.log it) or use it in an event handler it just jumps from 0 to 1
